Notebooks/LSST_Stack_matching_code.ipynb

```diff
@@ -0,0 +1,533 @@
+    "        calexp = butler.get('calexp', dataId=dataId)\n",
+    "        self._set_coord_range(calexp.getBBox(), calexp.getWcs())\n",
+    "\n",
+    "\n",
+    "class PatchSelector(RegionSelector):\n",
+    "    \"\"\"RegionSelector to use with skyMap patches, i.e., coadd data.\"\"\"\n",
+    "    def __init__(self, butler, tract, patch):\n",
+    "        super(PatchSelector, self).__init__()\n",
+    "        # Get the patch boundaries.\n",
+    "        skymap = butler.get('deepCoadd_skyMap')\n",
+    "        tractInfo = skymap[tract]\n",
+    "        patchInfo = tractInfo.getPatchInfo(eval(patch))\n",
+    "        self._set_coord_range(patchInfo.getOuterBBox(), tractInfo.getWcs())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Matching Experiments\n",
+    "\n",
+    "Now we have the tools we need, let's read in the Run 1.1p DRP catalog data and match it to the extragalactic catalog input. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a data butler for the repo.\n",
+    "repo = '/global/projecta/projectdirs/lsst/global/in2p3/Run1.1/output'\n",
+    "butler = dp.Butler(repo)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Sources or Objects?\n",
+    "\n",
+    "We can consider visit-level src catalog data, in which case we would provide a `dataId` to the butler with (`visit`, `raft`, `sensor`) ids; or we can consider coadd object data, for which we would provid a `dataId` with (`filter`, `tract`, `patch`) ids.\n",
+    "\n",
+    "Somewhat different flux models are available in the Run1.1p data for src catalogs versus coadd catalogs.  Ideally, we'd like to use CModel values for both, but settle here with `ext_photometryKron_KronFlux` values for the src catalog."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mag_max = 24.5\n",
+    "# we_are_matching = \"Sources\"\n",
+    "we_are_matching = \"Objects\"\n",
+    "\n",
+    "if we_are_matching == \"Sources\":\n",
+    "    # Get the src catalog for a selected visit, raft, and sensor:\n",
+    "    visit = 219976\n",
+    "    raft = '2,2'\n",
+    "    sensor = '1,1'\n",
+    "    title = 'Run1.1p, visit={}, raft={}, sensor={}'.format(visit, raft, sensor)\n",
+    "    dataId = dict(visit=visit, raft=raft, sensor=sensor)\n",
+    "    catalog = butler.get('src', dataId=dataId)\n",
+    "    calexp = butler.get('calexp', dataId=dataId)\n",
+    "    filter_ = calexp.getInfo().getFilter().getName()\n",
+    "    calib = calexp.getCalib()\n",
+    "    flux_model = 'ext_photometryKron_KronFlux'\n",
+    "    region_selector = CcdSelector(butler, visit, raft, sensor)\n",
+    "\n",
+    "else:\n",
+    "    # Get the coadd catalog for a selected filter, tract, and patch:\n",
+    "    filter_ = 'r'\n",
```

Comment:
This line had a typo in it - `filter` instead of `filter_`. Bug caught and fixed!